esx_service/utils/auth_data.py

```diff
@@ -538,18 +538,80 @@ def handle_upgrade_db_from_ver_1_0_to_ver_1_1(self):
         logging.error(error_msg)
         raise DbAccessError(self.db_path, error_msg)
 
+    def handle_upgrade_db_from_ver_1_1_to_ver_1_2(self):
+        """
+        Upgrade the db from version 1.1 to 1.2
+        In 1.1 the vms table had two columns namely vm_id and tenant_id
+        In 1.2 the vms table has three columns namely vm_id, tenant_id and vm_name
+        In the process of upgrading existing db from 1.1 to 1.2, we need to populate
+        the vm names for existing records. We try to populate them and keep None for vms
+        for which the name couldn't be found. The vmdk_ops admin code which tries to use
+        this vm names handles names which are None
+        """
+        try:
+            # Retrieve the existing records and populate vm names for vm ids
+            cur = self.conn.execute(
+                "SELECT * FROM vms"
+            )
+            vm_list = cur.fetchall()
+            # The name could be 'None', could not find the name for vm
+            # Preserve None in db and handle it while printing names
+            vms = [(vm[0], vmdk_utils.get_vm_name_by_uuid(vm[0]), vm[1]) for vm in vm_list]
+
+            # clean the existing records
+            self.conn.execute(
+                "DELETE FROM vms"
+            )
+
+            # Modify the vms table schema
+            self.conn.execute(
+                '''
+                    ALTER TABLE vms
+                    ADD COLUMN vm_name TEXT
+                '''
+            )
+
+            # Insert the new records in vms table
+            self.conn.executemany(
+                "INSERT INTO vms(vm_id, vm_name, tenant_id) VALUES (?, ?, ?)",
+                vms
+            )
+
+            # update the version info for the db
+            self.conn.execute(
+                "UPDATE versions SET major_ver = ?, minor_ver = ?", (DB_MAJOR_VER, DB_MINOR_VER)
+            )
+
+            self.conn.commit()
+
+            return None
+        except sqlite3.Error as e:
+            logging.error("Error %s when upgrading auth DB VMs table", e)
+            return str(e)
+
     def handle_upgrade_db(self):
         major_ver = 0
         minor_ver = 0
         error_msg, major_ver, minor_ver = self.get_db_version()
         if error_msg:
             logging.error("handle_upgrade_db: fail to get version info of auth-db, cannot do upgrade")
 
+        # TODO remove this upgrade. We only support one version below current
         if DB_MAJOR_VER == 1 and DB_MINOR_VER == 1:
             # DB version read from auth-DB is 1.0, upgrade is needed
             if major_ver == 1 and minor_ver == 0:
                 self.handle_upgrade_db_from_ver_1_0_to_ver_1_1()
 
+        if DB_MAJOR_VER == 1 and DB_MINOR_VER == 2:
+            if major_ver == 1 and minor_ver == 1:
+                return self.handle_upgrade_db_from_ver_1_1_to_ver_1_2()
+            else:
+                error_msg = "handle_upgrade_db: failed to upgrade auth-db schema"
```

Comment:
Could you change the error msg to say that "upgrade is not support for auth-db schema schema version {} to schema version {}"?